src/Client/Dialogs/SetupDialog.razor.cs

```diff
@@ -0,0 +1,43 @@
+﻿
+using Microsoft.AspNetCore.Components;
+using MudBlazor;
+
+namespace InterviewApp.Client.Dialogs;
+
+public partial class SetupDialog
+{
+    [CascadingParameter]
+    MudDialogInstance? MudDialog { get; set; }
+
+    [Parameter] 
+    public IList<string>? Categories { get; set; }
+
+    public IList<string>? SelectedCategories { get; set; } = new List<string>();
+
+    void Submit()
+        => MudDialog?.Close(DialogResult.Ok(SelectedCategories));
+
+    void ToggleCategory(string category)
+    {
+        if (Categories is null)
+        {
+            throw new InvalidOperationException($"{nameof(Categories)} is not initialized.");
+        }
+
+        if (SelectedCategories is null)
+        {
+            throw new InvalidOperationException($"{nameof(SelectedCategories)} is not initialized.");
+        }
+
+        if (Categories.Contains(category))
+        {
+            Categories.Remove(category);
+            SelectedCategories.Add(category);
+        }
+        else if (SelectedCategories.Contains(category))
```

Comment:
How about getting rid of the else-if clause for readability reasons? 😊 If you want only one condition to pass, you might do two seperate if statements each containing a return.

Imho it might improve readability and potential refactor 🌮